Test: Add a new global JustBeforeEach function to validate internet connectivity

eloycoto opened the pull request:
Add a new JustBeforeEach function that curl from host to google to
validate that internet connectivity is in there.

Fixes #5853

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/5860)
<!-- Reviewable:end -->

eloycoto committed:
Test: Add a new global JustBeforeEach function to validate internet connectivity.

Add a new JustBeforeEach function that curl from host to google to
validate that internet connectivity is in there.

Fixes #5853

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

ianvernon commented on test/test_suite_test.go:
```diff
@@ -241,6 +245,14 @@ var _ = BeforeAll(func() {
+	if node == nil {
+		return
+	}
+	node.Exec(helpers.CurlFail("-4 https://google.com")).ExpectSuccess(
```
Add a `By` here explaining that we are checking external connectivity before each test. Also would be helpful to add a comment explaining that this is done as a sanity check before doing external connectivity tests from Cilium-managed pods to determine if connectivity problems are due to infrastructural issues vs. regressions in Cilium itself.

eloycoto replied:
Fixed

eloycoto committed:
Test: Pre-flight check, added mtr output.

Added a mtr output when the test fails to be able to send to the host
provider.

If the test fails, the mtr to 8.8.8.8 wil happens.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
WIP: Testing in another datacenter

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

ianvernon commented on ginkgo.Jenkinsfile:
```diff
@@ -2,7 +2,7 @@
 pipeline {
     agent {
-        label 'baremetal'
+        label 'eloy'
```
Remove this.

eloycoto replied:
Sorry! WIP

eloycoto committed:
Add a 1.1.1.1 dns

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
